plugins/UFM_NDT_Plugin/ndt_file_creator.py

```diff
@@ -153,64 +170,78 @@ def parse_ibdiagnet_dump(net_dump_file_path):
-    exit(1)
-if not check_file_exist(IBDIAGNET_NET_DUMP_FILE):
-    print("ibdiagnet output file %s not exist" % IBDIAGNET_NET_DUMP_FILE)
-    exit(1)
-
-ibdiagnet_links, ibdiagnet_links_reverse, links_info_dict , kaka = parse_ibdiagnet_dump(IBDIAGNET_NET_DUMP_FILE)
-#{'end_dev': 'AJNA-LEAF-01',
-# 'end_port': '40',
-# 'start_dev': 'AJNA-ROOT-01',
-# 'start_port': '40',
-# 'unique_key': 'AJNA-ROOT-0140AJNA-LEAF-0140'}
-#rack #,U height,#Fields:StartDevice,StartPort,StartDeviceLocation,EndDevice,EndPort,EndDeviceLocation,U height_1,LinkType,Speed,_2,Cable Length,_3,_4,_5,_6,_7,State,Domain
-#,,SwitchX -  Mellanox Technologies,Port 26,,r-ufm64 mlx5_0,Port 1,,,,,,,,,,,,Active,In-Scope
-header="rack #,U height,#Fields:StartDevice,StartPort,StartDeviceLocation,EndDevice,EndPort,EndDeviceLocation,U height_1,LinkType,Speed,_2,Cable Length,_3,_4,_5,_6,_7,State,Domain\n"
-with open(OUTPUT_NDT_FILE_NAME, 'w') as topoconfig_file:
-    topoconfig_file.write(header)
-    for link in ibdiagnet_links:
-        line = ",,%s,Port %s,,%s,Port %s,,,,,,,,,,,,Active,In-Scope\n" % (link.start_dev, link.start_port, link.end_dev, link.end_port)
-        topoconfig_file.write(line)
-
-print("NDT file generation completed. Could be found at %s" % OUTPUT_NDT_FILE_NAME)
 
+    return ibdiagnet_links, links_list_disconnected
+
+
+def main():
+    '''
+    Main script function
+    '''
+        # arguments parser
+    parser = create_base_parser()
+    allocate_request_args(parser)
+    arguments = vars(parser.parse_args())
+    request_arguments = extract_request_args(arguments)
+    input_path = request_arguments['input_path']
+
+    if not input_path:
+        # run ibdiagnet command to create output
+        ibdiag_net_dump_file = IBDIAGNET_NET_DUMP_FILE
+        if not run_ibdiagnet():
+            print("Filed to run ibdiagnet command: %s" % IBDIAGNET_COMMAND)
+            exit(1)
+    else:
+        # use ibdiagnet dump
+        ibdiag_net_dump_file = os.path.join(input_path, NETDUMP_FILE_NAME)
+    # check if net dump file exist
+    if not check_file_exist(ibdiag_net_dump_file):
+        print("ibdiagnet output file %s not exist" % ibdiag_net_dump_file)
+        exit(1)
+    include_down_ports = (request_arguments['include_down_ports'] == "yes")
+    ibdiagnet_links, links_list_disconnected = parse_ibdiagnet_dump( ibdiag_net_dump_file,
+                                                             include_down_ports)
+    #rack #,U height,#Fields:StartDevice,StartPort,StartDeviceLocation,EndDevice,EndPort,EndDeviceLocation,U height_1,LinkType,Speed,_2,Cable Length,_3,_4,_5,_6,_7,State,Domain
+    #,,SwitchX -  Mellanox Technologies,Port 26,,r-ufm64 mlx5_0,Port 1,,,,,,,,,,,,Active,In-Scope
+    header="rack #,U height,#Fields:StartDevice,StartPort,StartDeviceLocation,EndDevice,EndPort,EndDeviceLocation,U height_1,LinkType,Speed,_2,Cable Length,_3,_4,_5,_6,_7,State,Domain\n"
+    with open(OUTPUT_NDT_FILE_NAME, 'w') as topoconfig_file:
```

Comment:
topoconfig_file --> ndt_output_file  (topoconfig_file is something else which is confusing...)
change the relevant references after that...

Reply:
changed